src/parser.rs

```diff
@@ -344,37 +344,37 @@ impl<'a> Parser<'a> {
     Ok(())
   }
 
-  fn alias(&mut self, alias_name: Token<'a>) -> CompilationResult<'a, ()> {
+  fn alias(&mut self, name: Token<'a>) -> CompilationResult<'a, ()> {
     // Make sure alias doesn't already exist
-    if self.aliases.contains_key(alias_name.lexeme) {
-      return Err(alias_name.error(DuplicateAlias {
-        alias: alias_name.lexeme, 
+    if let Some(alias) = self.aliases.get(name.lexeme) {
+      return Err(name.error(DuplicateAlias {
+        alias: alias.name, 
+        first: alias.line_number,
       }))
     }
 
     // Make sure the next token is of kind Name and keep it
-    let next = self.tokens.next().unwrap();
-    let recipe_name;
-    match next.kind {
-      Name => recipe_name = next,
-      _ => return Err(self.unexpected_token(&next, &[Name]))
-    }
+    let target = if let Some(next) = self.accept(Name) {
+      next.lexeme
+    } else {
+      let unexpected = self.tokens.next().unwrap();
+      return Err(self.unexpected_token(&unexpected, &[Name]));
+    };
 
-    // Make sure this recipe exists
-    if None == self.recipes.get(recipe_name.lexeme) {
-      return Err(alias_name.error(AliasingUnknownRecipe {
-        alias: alias_name.lexeme, recipe: recipe_name.lexeme
-      }))
+    // Make sure this is where the line or file ends without any unexpected tokens.
+    if let Some(token) = self.expect_eol() {
+      return Err(self.unexpected_token(&token, &[Eol, Eof, Comment]))
```

Comment:
I think that it can just be `&[Eol, Eof]`, instead of `&[Eol, Eof, Comment]`.

The list of tokens given to `unexpected_token` are purely advisory, and just give the users an idea of what they could have put instead. 

Although it is true that a comment can go here, since comments can go basically anywhere, it probably isn't necessary to note in the error message.